config/routes.rb

```diff
@@ -146,6 +146,13 @@
 
       scope module: :forum do
         resources :forums do
+          resources :topics do
+            post 'subscribe', on: :member
+            delete 'unsubscribe', on: :member
+            put 'set_lock', on: :member
```

Comment:
if you are thinking of the locked field, then it should be put, not patch. the field is an atom, there is nothing to patch.

I think of it as post because set is a verb, so it is an action. if it is a mutator, then it should be put.